src/components/templates/PartyMap/PartyMap.tsx

```diff
@@ -4,78 +4,122 @@ import { useParams } from "react-router-dom";
 import { RootState } from "index";
 import { createUrlSafeName } from "api/admin";
 
-import { PartyMapRoomData } from "types/PartyMapRoomData";
-import { PartyMapVenue } from "types/PartyMapVenue";
+import { Room } from "types/rooms";
+import { PartyMapVenue } from "types/venues";
 
-import { usePartygoers } from "hooks/users";
 import { useSelector } from "hooks/useSelector";
+import { useUser } from "hooks/useUser";
+import { usePartygoers } from "hooks/users";
+
+import { orderedVenuesSelector } from "utils/selectors";
+import { getCurrentTimeInUTCSeconds } from "utils/time";
+import { openRoomUrl } from "utils/url";
+import { trackRoomEntered } from "utils/useLocationUpdateEffect";
 
 import { Map, RoomModal } from "./components";
 
-import "./PartyMap.scss";
 import AnnouncementMessage from "components/molecules/AnnouncementMessage/AnnouncementMessage";
 import SparkleFairiesPopUp from "components/molecules/SparkleFairiesPopUp/SparkleFairiesPopUp";
 
+import "./PartyMap.scss";
+
 const partyMapVenueSelector = (state: RootState) =>
   state.firestore.ordered.currentVenue?.[0] as PartyMapVenue;
 
 export const PartyMap: React.FC = () => {
-  const [isRoomModalOpen, setIsRoomModalOpen] = useState(false);
-  const [selectedRoom, setSelectedRoom] = useState<
-    PartyMapRoomData | undefined
-  >();
-
-  const venue = useSelector(partyMapVenueSelector);
-  const usersInCamp = usePartygoers();
-
-  const attendances = useMemo(
-    () =>
-      usersInCamp
-        ? usersInCamp.reduce<Record<string, number>>((acc, value) => {
-            Object.keys(value.lastSeenIn).forEach((key) => {
-              acc[key] = (acc[key] || 0) + 1;
-            });
-            return acc;
-          }, {})
-        : {},
-    [usersInCamp]
-  );
+  const { user, profile } = useUser();
+  const currentVenue = useSelector(partyMapVenueSelector);
+  const venues = useSelector(orderedVenuesSelector);
 
-  const modalHidden = useCallback(() => {
-    setIsRoomModalOpen(false);
+  // TODO: can we just make this a boolean that is set from isTruthy(selectedRoom)?
+  const [isRoomModalOpen, setRoomModalOpen] = useState(false);
+  const [selectedRoom, setSelectedRoom] = useState<Room | undefined>();
```

Comment:
> can we just make this a boolean that is set from isTruthy(selectedRoom)?

Apparently the nuance of how this is used is used to neatly show/hide the `RoomModal` without it 'losing the data' as it's transitioning to hidden. Not sure if that is a big deal though?

Reply:
Can you use the `selectedRoom` instead of the state above to show/hide the modal?

Reply:
I thought about that as well, but then I realised that some of the little UI bugs i'm seeing at the moment (eg. the modal closing instantly rather than 'smoothly') is because I joined these concepts together, and when I unset `selectedRoom`, the modal doesn't render anything.

I had some thoughts about how I could fix this, but need to get the 🧠  back into this code to check if they'll work, or if I just need to revert this to being two separate things entirely.

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1080/commits/2c8cd7105dd5d01c7b58c1f7f9751c3fb7022ba5